bundles/org.openhab.core.io.rest.core/src/main/java/org/openhab/core/io/rest/core/internal/thing/ThingResource.java

```diff
@@ -291,13 +307,34 @@ public Response create(
     @Operation(operationId = "getThings", summary = "Get all available things.", security = {
             @SecurityRequirement(name = "oauth2", scopes = { "admin" }) }, responses = {
                     @ApiResponse(responseCode = "200", description = "OK", content = @Content(array = @ArraySchema(schema = @Schema(implementation = EnrichedThingDTO.class), uniqueItems = true))) })
-    public Response getAll(
+    public Response getAll(@Context Request request,
             @HeaderParam(HttpHeaders.ACCEPT_LANGUAGE) @Parameter(description = "language") @Nullable String language,
-            @QueryParam("summary") @Parameter(description = "summary fields only") @Nullable Boolean summary) {
+            @QueryParam("summary") @Parameter(description = "summary fields only") @Nullable Boolean summary,
+            @DefaultValue("false") @QueryParam("staticDataOnly") @Parameter(description = "provides a cacheable list of values not expected to change regularly and checks the If-Modified-Since header") boolean staticDataOnly) {
         final Locale locale = localeService.getLocale(language);
 
         Stream<EnrichedThingDTO> thingStream = thingRegistry.stream().map(t -> convertToEnrichedThingDTO(t, locale))
                 .distinct();
+
+        if (staticDataOnly) {
+            if (cacheableListLastModified != null) {
+                Response.ResponseBuilder responseBuilder = request.evaluatePreconditions(cacheableListLastModified);
+                if (responseBuilder != null) {
+                    // send 304 Not Modified
+                    return responseBuilder.build();
+                }
+            } else {
+                cacheableListLastModified = Date.from(Instant.now().truncatedTo(ChronoUnit.SECONDS));
+            }
+
+            CacheControl cc = new CacheControl();
+            cc.setMustRevalidate(true);
+            cc.setPrivate(true);
+            thingStream = dtoMapper.limitToFields(thingStream, "UID,label,bridgeUID,thingTypeUID,location,editable");
```

Comment:
@ghys @J-N-K Is is possible that channels change without a notification to the registry? 

If not, we should add channels here to fix https://github.com/openhab/openhab-webui/issues/1996 - the alternative would be to not use the cached list there.

Reply:
@J-N-K Sorry for pinging again, but I think you can answer my question abovey

Reply:
Technically it might be possible, but I would consider it a bug in the binding. You might be able to edit the thing, but if you do it right (using `ThingBuilder thingBuilder = editThing();` and `updateThing(thingBuilder.build);`) you'll get a notification about an updated thing.

Reply:
> @ghys @J-N-K Is is possible that channels change without a notification to the registry? 
> 
> If not, we should add channels here to fix https://github.com/openhab/openhab-webui/issues/1996 - the alternative would be to not use the cached list there.

it's IMO way better to just change the channeltype-picker in the UI to not use cached lists. There is no need to add channels (a huge chunk of the response payload, some things have dozens if not hundreds of channels), they're not used apart from that picker and there is only one instance of them in the entire UI.